src/client.cpp

```diff
@@ -217,17 +218,16 @@ void Client::loadModSubfolder(const std::string &mod_name, const std::string &mo
 
 void Client::initMods()
 {
-	m_script->loadModFromMemory(BUILTIN_MOD_NAME);
-
 	// If modding is not enabled, don't load mods, just builtin
 	if (!m_modding_enabled) {
 		return;
 	}
 
 	// Load and run "mod" scripts
-	for (std::vector<ModSpec>::const_iterator it = m_mods.begin();
-			it != m_mods.end(); ++it) {
-		const ModSpec &mod = *it;
+	for (ModSpec mod : m_mods) {
```

Comment:
it's only a proof of concept it's not indented to merged which out major improvements.